crates/ruff_python_formatter/src/other/parameters.rs

```diff
@@ -102,7 +102,15 @@ impl FormatNodeRule<Parameters> for FormatParameters {
             dangling.split_at(parenthesis_comments_end);
 
         let format_inner = format_with(|f: &mut PyFormatter| {
-            let separator = format_with(|f| write!(f, [token(","), soft_line_break_or_space()]));
+            let separator = format_with(|f: &mut PyFormatter| {
+                token(",").fmt(f)?;
+
+                if f.context().node_level().is_parenthesized() {
```

Comment:
testing for `self.parentheses == ParameterParentheses::Never` is, unfortunately, not sufficient because it doesn't handle comments:

```python
a = [
	lambda x # comment
		x 
		# comment
	: b
]
```

This uses the `Never` layout but the lambda has comments. The example also shows that testing whether the lambda expression itself is parenthesized isn't sufficient.